README.md

```diff
@@ -20,95 +24,252 @@ var marked = transformer(require('jstransformer-marked'));
 
 var options = {};
 var res = marked.render('Some **markdown**', options);
-// => {body: 'Some <strong>markdown</strong>', dependencies: []}
+//=> {body: 'Some <strong>markdown</strong>', dependencies: []}
 ```
 
 This gives the same API regardless of the jstransformer passed in.
 
 ## API
+A transformer, once normalised using this module, will implement the following methods. Note that if the underlying transformer cannot be used to implement the functionality, it may ultimately just throw an error.
 
-A transformer, once normalised using this module, will implement the following methods.  Note that if the underlying transformer cannot be used to implement the functionality, it may ultimately just throw an error.
 
-### Returned object from `.render*`
+### Returned object from `.compile*`
+```js
+{fn: Function, dependencies: Array.<String>}
+```
+
+ - `fn` represents the render function, which also accept `locals`
+ - `dependencies` is an array of files that were read in as part of the compile process (or an empty array if there were no dependencies)
+
+
+### .compile
+Compile a given string and return an object.
+
+**Params**
+- `<str>` **{String}** string to compile
+- `[options]` **{Object}** optional options to pass to
+- `returns` **{Object}**
+
+**Requires the underlying transform to implement one of following methods**
```

Comment:
I would suggest breaking this into a concise heading (e.g. "Implementation Methods") followed by a non-bold paragraph.